CraftTweaker2-API/src/main/java/crafttweaker/api/block/BlockStateMatcherOr.java

```diff
@@ -0,0 +1,51 @@
+package crafttweaker.api.block;
+
+import crafttweaker.util.ArrayUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class BlockStateMatcherOr implements IBlockStateMatcher {
+
+    private final IBlockStateMatcher[] elements;
+
+    public BlockStateMatcherOr(IBlockStateMatcher[] elements) {
+        this.elements = elements;
+    }
+
+    public BlockStateMatcherOr(IBlockStateMatcher a, IBlockStateMatcher b) {
+        this.elements = new IBlockStateMatcher[]{a, b};
+    }
+
+    @Override
+    public boolean matches(IBlockState blockState) {
+        for (IBlockStateMatcher matcher : elements) {
+            if (matcher.matches(blockState)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public IBlockStateMatcher allowValuesForProperty(String name, String... values) {
+        return null;
```

Comment:
Some sort of message should at least be returned here. I'd rather not see null and have it just return itself. But at least a message mentioning why it's null.

Reply:
Would it be worthwhile here to return a `BlockStateMatcherOr` with no elements so it would functionally match nothing? That would let you still chain to it, etc.